extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpAuthenticator.java

```diff
@@ -93,25 +101,35 @@ public Uni<SecurityIdentity> attemptAuthentication(RoutingContext routingContext
                 ? pathMatchingPolicy.getAuthMechanismName(routingContext)
                 : null;
         Uni<HttpAuthenticationMechanism> matchingMechUni = findBestCandidateMechanism(routingContext, pathSpecificMechanism);
+        final Uni<SecurityIdentity> result;
         if (matchingMechUni == null) {
-            return createSecurityIdentity(routingContext);
-        }
-
-        return matchingMechUni.onItem()
-                .transformToUni(new Function<HttpAuthenticationMechanism, Uni<? extends SecurityIdentity>>() {
-
-                    @Override
-                    public Uni<SecurityIdentity> apply(HttpAuthenticationMechanism mech) {
-                        if (mech != null) {
-                            return mech.authenticate(routingContext, identityProviderManager);
-                        } else if (pathSpecificMechanism != null) {
-                            return Uni.createFrom().optional(Optional.empty());
+            result = createSecurityIdentity(routingContext);
+        } else {
+            result = matchingMechUni.onItem()
+                    .transformToUni(new Function<HttpAuthenticationMechanism, Uni<? extends SecurityIdentity>>() {
+
+                        @Override
+                        public Uni<SecurityIdentity> apply(HttpAuthenticationMechanism mech) {
+                            if (mech != null) {
+                                return mech.authenticate(routingContext, identityProviderManager);
+                            } else if (pathSpecificMechanism != null) {
+                                return Uni.createFrom().optional(Optional.empty());
+                            }
+                            return createSecurityIdentity(routingContext);
                         }
-                        return createSecurityIdentity(routingContext);
-                    }
-
-                });
 
+                    });
+        }
+        if (fireAuthenticationEvent) {
```

Comment:
Probably makes sense to fire for positive results too

Reply:
agreed, I answer this little above, didn't do it to limit downsides for those not interested in auth success. let's discuss in the other thread